File: medium/73_set_matrix_zeroes.java
class Solution {
    public void setZeroes(int[][] matrix) {
        
        int n = matrix.length;
        int m = matrix[0].length;
        Set<Integer> xList = new HashSet<>();
        Set<Integer> yList = new HashSet<>();
        for(int i = 0;i < n;i++){
            for(int j = 0;j<m;j++){
                if(matrix[i][j]==0){
                    xList.add(i);
                    yList.add(j);
                }
            }
        }

         for(int i = 0;i < n;i++){
            for(int j = 0;j<m;j++){
                if(xList.contains(i) || yList.contains(j)){
                    matrix[i][j]=0;
                }
            }
         }

    }
}

File: medium/73_set_matrix_zeroes.cpp
class Solution {
public:
    void setZeroes(vector<vector<int>>& matrix) {
        int m=matrix.size();
        int n=matrix[0].size();
        vector<int> col(m,-1);
        vector<int> row(n,-1);
        for(int i=0;i<m;i++){
            for(int j=0;j<n;j++){
                if(matrix[i][j]==0){
                    col[i]=0;
                    row[j]=0;
                }
            }
        }
        for(int i=0;i<m;i++){
            for(int j=0;j<n;j++){
                if(col[i]==0 || row[j]==0){
                    matrix[i][j]=0;
                }
                cout<<matrix[i][j];
            }
            cout<<endl;
        }
    }
};